src/components/ButtonGroup/components/Item/Item.tsx

```diff
@@ -9,39 +9,18 @@ export interface ItemProps {
   button: React.ReactElement<ButtonProps>;
 }
 
-interface State {
-  focused: boolean;
-}
-
-export class Item extends React.PureComponent<ItemProps, State> {
-  state: State = {focused: false};
-
-  render() {
-    const {button} = this.props;
-    const {focused} = this.state;
-
-    const className = classNames(
-      styles.Item,
-      focused && styles['Item-focused'],
-      button.props.plain && styles['Item-plain'],
-    );
-
-    return (
-      <div
-        className={className}
-        onFocus={this.handleFocus}
-        onBlur={this.handleBlur}
-      >
-        {button}
-      </div>
-    );
-  }
-
-  private handleFocus = () => {
-    this.setState({focused: true});
-  };
-
-  private handleBlur = () => {
-    this.setState({focused: false});
-  };
-}
+export const Item = React.memo(function Item({button}: ItemProps) {
+  const [focused, setFocused] = useToggle(false);
+
+  const className = classNames(
+    styles.Item,
+    focused && styles['Item-focused'],
+    button.props.plain && styles['Item-plain'],
+  );
+
+  return (
+    <div className={className} onFocus={setFocused} onBlur={setFocused}>
```

Comment:
Will this potentially cause oddness if you try and focus something that is already focussed, or blur something that is already blured? (e.g. if it's already focused causing it to focus again will cause it to toggle to the unfocused state instead of remaining focused)

`useForcableToggle()` and its options for "always set to true" and "always set to false" would be better here.